code/datums/status_effects/debuffs/disgust.dm

```diff
@@ -0,0 +1,104 @@
+/**
+ * The disgust status effect.
+ * Slowly decreases in disgust_value over time, causing effects based on that value.
+ * Uses a lot of copypaste from the drunk status effect.
+ */
+/datum/status_effect/disgust
+	id = "disgust"
+	tick_interval = 2 SECONDS
+	status_type = STATUS_EFFECT_UNIQUE
+	remove_on_fullheal = TRUE
+	/// The level of disgust we are currently at.
+	var/disgust_value = 0
+
+/datum/status_effect/disgust/on_creation(mob/living/new_owner, disgust_value = 0)
+	. = ..()
+	//this is pointless for non-carbons
+	if(!iscarbon(new_owner))
+		qdel(src)
+		return
```

Comment:
Put this in `on_apply()` instead, serves the same purpose
```dm
if(!iscarbon(owner)
    return FALSE
```

Reply:
thought so, good